bin/flutter.bat

```diff
@@ -47,6 +47,18 @@ IF NOT EXIST "%flutter_root%\.git" (
 REM Ensure that bin/cache exists.
 IF NOT EXIST "%cache_dir%" MKDIR "%cache_dir%"
 
+REM If the cache still doesn't exist, fail with an error that we probably don't have permissions.
+IF NOT EXIST "%cache_dir%" (
+  ECHO Error: Unable to create cache directory at
+  ECHO            %cache_dir%
+  ECHO.
+  ECHO        This may because you do not have sufficient permissions to write to
+  ECHO        this folder and is often caused by installing Flutter into a
+  ECHO        folder like C:\Program Files and then running as a non-Admin.
```

Comment:
This may encourage users to run flutter ad admin, which is something we shouldn't do IMHO. Actually, on Linux we explicitly disallow that. Can we rephrase this to something that doesn't push users towards running things as admin? E.g. "This may be because flutter doesn't have write permissions for %FLUTTER_ROOT%. Consider moving flutter to a writable path, e.g. your home directory."